### PR TITLE
Async select

### LINKED ISSUE #12427: Async select editor
### Context

In #12386, we discovered that the component is rendered twice if the user is editing the form. This is because of the use of the `setSchema` we do, with the fact we need to wait to the queryGeometryModel to be fetched before to request all the regions. This is working but is inefficient, two renders and duplication of some requests, and annoying, it provokes some flickering in the UI.

So in order to fix this properly, we need to revamp the select editor to make it async-aware, in charge of auto render when the collection is fetched, avoiding the whole form's render.

Select editor is extensively used across the Builder, that's the reason to tackle this issue independently.

### PR DESCRIPTION
**Pull request by nobuti**
This PR fixes #12427 and #12426.

### CODE REVIEW

File: lib/assets/core/javascripts/cartodb3/components/form-components/editors/select/select-list-view.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      typeLabel: this.options.typeLabel,
+      itemTemplate: this._itemTemplate,
+      itemView: this._itemView,
+      position: this.options.position,
+      searchPlaceholder: this.options.searchPlaceholder
+    });
+
+    this.addView(this._listView);
+    this._listView.show();
+    this.el.appendChild(this._listView.el);
```
Comment (xavijam) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/select/select-list-view.js]: Don't we need to render it when it is appended?

File: lib/assets/core/javascripts/cartodb3/components/form-components/editors/select/select-list-view.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  _getStatus: function () {
+    if (this._collection.isAsync()) {
+      return this._collection.stateModel.get('state');
+    } else {
+      return 'fetched';
+    }
+  },
+
+  _destroyBinds: function () {
+    this.stopListening(this._collection);
```
Comment (xavijam) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/select/select-list-view.js]: Let's not forget to add `this._removeESCandClickListeners()` like in the base editor.

Comment (nobuti) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/select/select-list-view.js]: But that view doesn't have that method. The editor is the parent, not the list view.

Comment (xavijam) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/select/select-list-view.js]: Ahhhhhh, 👍

**Comment by xavijam** [comment]
Found with @nobuti that:
-  search is not working properly (just tested in DO region editor).
- when timespan is selected, it is not rendered displaying the selected item.

**Comment by nobuti** [comment]
Both issues are fixed. Also, the slider component doesn't trigger change until the user releases the dragger. This is enqueued to deploy due `Jenkins is going to shut down`™.

**Comment by xavijam** [comment]
Working like a charm now, should I test the rest of the places where the select component is being used?
I've found one minor thing with the boundaries slider, when there is only one value:

slider

**Comment by xavijam** [comment]
Reviewing all available analysis + places where select is being used.

**Comment by xavijam** [comment]
Analyses reviewed:

- DO: 👌  (several custom selects included).
- Filter by layer: 👌  (source select, standard selects).
- Subsample: 👌 , no selects :joy:.
- Georeference: 👌 , tested in most "complicated" one, street-by-address, where there are standard and custom selects.
- Intersection: 👌 .
- AOI: 👌 , no selects.
- Merge: 👌 , custom selects included.
- Centroid: 👌 , with enabler-selector with select, pretty nice.
- Convex hull: 👌 , with operation and enabled-selector, 💃 .
- Filter by column: value: 👌 .
- Filter points in polygons: 👌 .
- Connect with lines: 👌 .
- Cluster: 👌 , no selects.
- Markov: 👌 .
- Find nearest: 👌 .

**Comment by xavijam** [comment]
Frontend tests passing, merging 💋 .

**Pull request merged**